src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -781,8 +812,21 @@ public void prepareForTranslogRecovery() throws ElasticsearchException {
         // we disable deletes since we allow for operations to be executed against the shard while recovering
         // but we need to make sure we don't loose deletes until we are done recovering
         config.setEnableGcDeletes(false);
-        createNewEngine();
         recoveryState.setStage(RecoveryState.Stage.TRANSLOG);
+        createNewEngine(skipTranslogRecovery);
```

Comment:
@bleskes not sure what you mean the TRANSLOG phase starts once the ctor is called?